core/src/broadcast_stage/standard_broadcast_run.rs

```diff
@@ -190,6 +190,15 @@ impl StandardBroadcastRun {
             &receive_results.entries,
             last_tick_height == bank.max_tick_height(),
         );
+        //Insert the first shred so blocktree stores that the leader started this block
+        //This must be done before the blocks are sent out over the wire.
+        if data_shreds.len() > 0 && data_shreds[0].index() == 0 {
+            let first = vec![data_shreds[0].clone()];
```

Comment:
Wouldn't this cause the first shred to be inserted twice? First time here, and then thru blocktree_sender.send()?

Reply:
I thnk right now b/c the is_trusted flag iis set, it will jsut overwrite it the seccond tiime XD

Reply:
Sounds like a problem waiting to happen..

Reply:
@pgarg66 why?  I would expect the database to be sound if the same data is inserted twice.  that seems like a simple requirement.

Reply:
I think it's less about whether the db can handle it and more about a redundant call to insert the same data. Nbd I think since we'll hopefully get rid of this later.

Reply:
@sagar-solana what is the concern?  I can't imagine this ever being noticeable for performance.